command/setup_consul.go

```diff
@@ -0,0 +1,736 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package command
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+// Ensure SetupConsulCommand satisfies the cli.Command interface.
+var _ cli.Command = &SetupConsulCommand{}
+
+//go:embed asset/consul-wi-default-auth-method-config.json
+var consulAuthConfigBody []byte
+
+//go:embed asset/consul-wi-default-policy.hcl
+var consulPolicyBody []byte
+
+const (
+	consulAuthMethodServicesName = "nomad-services"
+	consulAuthMethodServicesDesc = "Login method for Nomad services using workload identities"
+	consulAuthMethodTasksName    = "nomad-tasks"
+	consulAuthMethodTaskDesc     = "Login method for Nomad tasks using workload identities"
+	consulRoleTasks              = "role-nomad-tasks"
+	consulPolicyName             = "policy-nomad-tasks"
+	consulNamespace              = "nomad-workloads"
+	consulAud                    = "consul.io"
+)
+
+type SetupConsulCommand struct {
+	Meta
+
+	// client is the Consul API client shared by all functions in the command
+	// to reuse the same connection.
+	client    *api.Client
+	clientCfg *api.Config
+
+	jwksURL string
+
+	consulEnt bool
+	autoYes   bool
+
+	// store IDs of objects we may need to cleanup
+	policyID       string
+	bindingRuleIDs []string
+}
+
+// Help satisfies the cli.Command Help function.
+func (s *SetupConsulCommand) Help() string {
+	helpText := `
+Usage: nomad setup consul [options]
+
+  This command sets up Consul for allowing Nomad workloads to authenticate
+  themselves using Workload Identity.
+
+  This command requires acl:write permissions for Consul and respects
+  CONSUL_HTTP_TOKEN, CONSUL_HTTP_ADDR, and other Consul-related
+  environment variables as documented in
+  https://developer.hashicorp.com/nomad/docs/runtime/environment#summary.
+
+  WARNING: This command is an experimental feature and may change its behavior
+  in future versions of Nomad.
+
+Setup Consul options:
+
+  -jwks-url <url>
+    URL of Nomad's JWKS endpoint contacted by Consul to verify JWT
+    signatures. Defaults to http://localhost:4646/.well-known/jwks.json.
+
+  -y
+    Automatically answers "yes" to all the questions, making the setup
+    non-interactive. Defaults to "false".
+
```

Comment:
It could be useful to have a flag like `-cleanup` or `-remove` that just calls `handleNo()`. Otherwise it's impossible to remove things once the setup completes because it just run though automatically 😅

Reply:
implemented in [599150e](https://github.com/hashicorp/nomad/pull/18820/commits/599150edb88dd997338d83c51c719014768a12fc)